hack/e2e-openstack.sh

```diff
@@ -1,14 +1,7 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -euo pipefail
+# TODO(stephenfin): This is legacy from when the OpenStack e2e tests lived
+# elsewhere. We should remove this script and update the CI jobs to call the
+# Makefile target like everyone else
 
-echo "Running e2e-openstack.sh"
-
-unset GOFLAGS
-tmp="$(mktemp -d)"
-
-echo "cloning github.com/openshift/cluster-api-provider-openstack at branch '$PULL_BASE_REF'"
-git clone --single-branch --branch="$PULL_BASE_REF" --depth=1 "https://github.com/openshift/cluster-api-provider-openstack.git" "$tmp"
-
-echo "running cluster-api-provider-openstack's: make e2e"
-exec make -C "$tmp/openshift" e2e
+exec make e2e
```

Comment:
Yeah I agree we should get rid of this and just invoke `make e2e` in the CI job config.
This is fine for this PR though, we can follow up with a second PR to remove this :)

Reply:
I'll address this TODO as soon as the PR is in :ok_hand:

Reply:
TY